app.py

```diff
@@ -185,17 +186,37 @@ def set_frame_range(self, engine, in_frame, out_frame):
 
         elif engine == "tk-nuke":
             import nuke
-
+            root = nuke.root()
+            lock_range = root["lock_range"]
             # unlock
-            locked = nuke.root()["lock_range"].value()
+            locked = lock_range.value()
+            change_msg = 'from %s-%s to %s-%s' % (prev_in, prev_out, in_frame, out_frame)
             if locked:
-                nuke.root()["lock_range"].setValue(False)
+                ret = nuke.ask("Frame range is currently locked,\n"
+                               "OK to update frame range %s ?" % change_msg)
+                if not ret:
+                    return
+            else:
+                p = nuke.Panel("Shall lock ?")
+                p.addMultilineTextInput("info",
+                                        "Frame range is going to be updated \n\n"
+                                        "%s,\n\n"
+                                        "please choose what to do.." % change_msg)
+                p.addButton('Reject')  # 0
+                p.addButton('Accept')  # 1
+                p.addButton('Accept but then LOCK the frame range')  # 2
```

Comment:
Minor: To make it shorter: `Accept and lock`. I don't think it is necessary to re-mention `frame range` as this dialog is already talking about frame range.